### PR TITLE
HTML-768: In HFE-UI, support form versions when loading via definitio…

### PR DESCRIPTION
**Pull request by mogoodrich**
…nUIResource

### CODE REVIEW

**Comment by mogoodrich** [comment]
@mseaton  note that I just put  in TODO in the flowsheet stuff... I didn't give it much thought, but lmk if you think we should deal with this now...

File: api/src/main/java/org/openmrs/module/htmlformentryui/HtmlFormUtil.java [hunk 1]
```diff
@@ -55,6 +69,19 @@ public static HtmlForm getHtmlFormFromUiResource(ResourceFactory resourceFactory
         return getHtmlFormFromResourceXml(formService, htmlFormEntryService, xml);
     }
 
+
+    // the new method above with "encounter" is preferred if an encounter is available, see: https://issues.openmrs.org/browse/HTML-768
+    public static HtmlForm getHtmlFormFromUiResource(ResourceFactory resourceFactory, FormService formService, HtmlFormEntryService htmlFormEntryService, String providerAndPath) throws IOException {
+        // using new Encounter() is a bit of a hack, used for a proper method signature hack; safe because the called method
+        // perform a null check on encounter.getForm()
+        return getHtmlFormFromUiResource(resourceFactory, formService, htmlFormEntryService, providerAndPath, new Encounter());
```
Comment (mseaton) [inline api/src/main/java/org/openmrs/module/htmlformentryui/HtmlFormUtil.java:77]: Don't love the "new Encounter()".  Could you achieve the same effect by doing "(Encounter)null" ?

File: omod/src/main/java/org/openmrs/module/htmlformentryui/page/controller/htmlform/FlowsheetPageController.java [hunk 2]
```diff
@@ -155,6 +155,7 @@ public void controller(@RequestParam(value="patientId", required=false) Patient
      */
     protected HtmlForm getHtmlFormFromResource(String resource, ResourceFactory rf, FormService fs, HtmlFormEntryService hfs) {
         try {
+            // TODO figure out how flowsheets might work with: https://issues.openmrs.org/browse/HTML-768
```
Comment (mseaton) [inline omod/src/main/java/org/openmrs/module/htmlformentryui/page/controller/htmlform/FlowsheetPageController.java:158]: Seems fine for now.

**Review by mseaton — APPROVED** [review]
LGTM @mogoodrich .  One comment to check out, but overall looks good.

File: api/src/main/java/org/openmrs/module/htmlformentryui/HtmlFormUtil.java (hunk 1 above)
Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/htmlformentryui/HtmlFormUtil.java]: Oh, cool, I never realized you could do that... it appears to work, thanks!

**Pull request merged**